Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3273.cs

```diff
@@ -22,6 +28,128 @@ public class SortableListView : ListView
 		{
 		}
 
+		[Preserve(AllMembers = true)]
+		public class AsyncCollection : IList, INotifyCollectionChanged, IEnumerator
+		{
+			List<object> _hash;
+			int _index;
+			int _count;
+			int _latency;
+
+			public AsyncCollection(int count)
+			{
+				_latency = 50;
+				_count = count;
+				_hash = Enumerable.Range(0, count).Select(x => (object)null).ToList();
+				_index = 0;
+			}
+
+			public int LoadedItems { get; private set; }
+
+			object SimulateAsyncGetItem(int i)
+			{
+				if (_hash[i] != null)
+					return _hash[i];
+
+				Thread.Sleep(_latency);
```

Comment:
Isn't this blocking the UIThread?

Reply:
Hm. Would it be fair to say this PR is trying to mitigate the effect of having the UIThread block on network traffic (loading from a slow remote server)? If so then I'd say while adding logic to prevent wrapping into an ObservableCollection may somehow mitigate UIThread blocking I'd suggest swapping to a background thread is a better overall solution. 

Is there a reason why a user couldn't swap to a background thread to do network traffic? Would that approach be slower (or impossible) in some cases and so justify using the optimization this PR is providing?

Reply:
I think the implementation of collection in this test is not so important. The fact is that you only get items from the user collection when they are needed.
Latency has been added to bring the test to real user conditions.